mdtraj/core/topology.py

```diff
@@ -641,13 +645,23 @@ def from_dataframe(
                     atom["resSeq"],
                     atom["segmentID"],
                 )
-
+            fc = atom["formal_charge"]
+            if fc is not None:
```

Comment:
Instead of having these if statements, you could cast to `Int64` before iterating over the atoms (with a try/except block and `astype` I mentioned above). I'd advise raising a `ValueError` in the `except`. I believe when casting to `Int64` it will ignore NaNs and also raise an error if a value that is not an integer is encountered. It will allow this to be done all at once (probably must faster than checking individual values) and will also be cleaner in the code.